source/common/network/address_impl.cc

```diff
@@ -130,7 +141,13 @@ int Ipv4Instance::connect(int fd) const {
 }
 
 int Ipv4Instance::socket(SocketType type) const {
-  return ::socket(AF_INET, flagsFromSocketType(type), 0);
+  int fd = ::socket(AF_INET, flagsFromSocketType(type), 0);
+#ifdef __APPLE__
+  if (fd != -1) {
```

Comment:
is there any point in doing fd != -1 check here? (Same in other places). Will just crash anyway in RELEASE_ASSERT.